dask/array/core.py

```diff
@@ -1247,8 +1252,22 @@ def chunksize(self):
     def dtype(self):
         return self._meta.dtype
 
+    @property
+    def _chunks(self):
+        return self.__chunks
+
+    @_chunks.setter
+    def _chunks(self, chunks):
+        self.__chunks = chunks
+
+        # When the chunks changes the cached properties that was dependent
+        # on it needs to be deleted:
+        for v in ["shape"]:
+            if v in self.__dict__:
```

Comment:
`.chunks` is already a property.  It feels strange to have a property of a property.  Maybe this logic can be rolled into the existing property definition?

Reply:
I had the same reaction, but then I saw that the setter for `chunks` just prompts people to not set chunks directly, so it kind of makes sense to separate this out.

Reply:
OK.  That makes  sense

Reply:
I agree, the problem was that self._chunks was used on so many other places that i was afraid i would miss something. And putting effort replacing on a larger design change that you maybe would instantly reject was also a factor.

Another thing is it is currently not allowed to set via self.chunks, so that property is effectively only a getter. And this trick requires a setter.